### PR TITLE
feat: add `includes` method to `array/fixed-endian-factory`

### LINKED ISSUE #3145: [RFC]: add `includes` method to `array/fixed-endian-factory`
### Description

This RFC proposes adding support for a `includes` method to [`@stdlib/array/fixed-endian-factory`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/fixed-endian-factory).

This method is the equivalent of the [`TypedArray.prototype.includes`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/includes) method, but for fixed-endian typed arrays.

When adding support, the following tasks should be completed:

1. The `./lib/main.js` file should be updated with the added method.
2. A new test file should be created which includes unit tests testing against expected behavior.
3. Two benchmark files should be added: one measuring baseline performance and another measuring performance as the array length increases.
4. The README should be updated to include documentation for the added method.

To provide a concrete example of what a PR adding the desired method should contain, see https://github.com/stdlib-js/stdlib/commit/956a4624c788689b1bca285856b987ea3aa32eb6, which is a commit adding `forEach` support. This should provide an idea of what is expected.

For an example of specialized typed arrays having customized methods, see

- [`@stdlib/array/bool`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/bool)
- [`@stdlib/array/complex128`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/complex128)
- [`@stdlib/array/complex64`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/complex64)

While the method proposed in this RFC will likely **not** match the implementations found in the above packages, those packages should provide a conceptual idea of what is desired. Do **not** simply copy-paste the code found in those packages.

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please do **not** submit LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `[RFC]:`.

### PR DESCRIPTION
**Pull request by ahmad-kashkoush**
Resolves #3145 .

## Description

> What is the purpose of this pull request?

This pull request:

-   support includes method to [@stdlib/array/fixed-endian-factory](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/fixed-endian-factory).

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3145 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by headlessNode** [comment]
@ahmad-kashkoush the title should start with "feat:"

**Comment by ahmad-kashkoush** [comment]
> @ahmad-kashkoush the title should start with "feat:"

Done, may you review PR?

File: lib/node_modules/@stdlib/array/fixed-endian-factory/README.md
Comment (headlessNode) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/README.md]:
```suggestion
Returns `true` if the specified `searchElement` is present in the array.
```

File: lib/node_modules/@stdlib/array/fixed-endian-factory/README.md
Comment (headlessNode) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/README.md]:
```suggestion
If the specified `searchElement` is not present in the array, the method returns `false`.
```

File: lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js
Comment (headlessNode) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js]:
```suggestion
	* Returns `true` if the specified `searchElement` is present in the array.
```

File: lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js
Comment (headlessNode) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js]:
```suggestion
	* @returns {boolean} boolean indicating whether the `searchElement` is present in the array
```

**Review by headlessNode — CHANGES_REQUESTED** [review]
@ahmad-kashkoush A few suggestions regarding the documentation, otherwise looks good. A more thorough review will be done by the maintainers. Also you should remove the parenthesis after `includes` in the title.

File: lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js
Comment (ahmad-kashkoush) [reply lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js]: ok,  thanks

**Comment by ahmad-kashkoush** [comment]
Good evening, @kgryte may you this a review

File: lib/node_modules/@stdlib/array/fixed-endian-factory/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/README.md]:
```suggestion
Returns a boolean indicating whether an array includes a provided value.
```

File: lib/node_modules/@stdlib/array/fixed-endian-factory/README.md
Comment (kgryte) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/README.md]:
```suggestion

idx = arr.includes( 5.0 );
```

File: lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js
Comment (kgryte) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js:679]:
```suggestion

	/**
```

File: lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js
Comment (kgryte) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js]:
```suggestion
	* Returns a boolean indicating whether an array includes a provided value.
```

File: lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js
Comment (kgryte) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js]:
```suggestion
	* @param {*} searchElement - search element
```

File: lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js
Comment (kgryte) [inline lib/node_modules/@stdlib/array/fixed-endian-factory/lib/main.js]:
```suggestion
	* @returns {boolean} boolean indicating whether an array includes a provided value
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @ahmad-kashkoush!

**Comment by kgryte** [comment]
@ahmad-kashkoush Would you mind resolving the merge conflicts? Once those are resolved, we can merge. Also, so that methods are listed in alphabetical order, `includes` should come before `indexOf` in both the README and `main.js` file.

**Pull request merged**
